mri/test/test_wavelet_adjoint.py

```diff
@@ -80,6 +85,40 @@ def test_Wavelet3D_PyWt(self):
             numpy.testing.assert_allclose(x_d, x_ad, rtol=1e-6)
         print(" Wavelet3 adjoint test passes")
 
+    def test_Wavelet_UD_2D(self):
+        """Test the adjoint operation for Undecimated wavelet
+        """
+        for i in range(self.max_iter):
+            print("Process Wavelet Undecimated test '{0}'...", i)
+            wavelet_op = WaveletUD2(nb_scale=4)
+            Img = (numpy.random.randn(self.N, self.N) +
+                   1j * numpy.random.randn(self.N, self.N))
+            f_p = wavelet_op.op(Img)
+            f = (numpy.random.randn(*f_p.shape) +
+                 1j * numpy.random.randn(*f_p.shape))
+            I_p = wavelet_op.adj_op(f)
+            x_d = numpy.vdot(Img, I_p)
+            x_ad = numpy.vdot(f_p, f)
+            numpy.testing.assert_allclose(x_d, x_ad, rtol=1e-6)
+        print("Undecimated Wavelet 2D adjoint test passes")
+
+    def test_Wavelet_UD_2D_Multichannel(self):
+        """Test the adjoint operation for Undecmated wavelet Transform in
+        multichannel case"""
+        for i in range(self.max_iter):
+            print("Process Wavelet Undecimated test '{0}'...", i)
+            wavelet_op = WaveletUD2(nb_scale=4, multichannel=True, n_cpu=2)
```

Comment:
Is it executing properly on different cpus in Travis? I mean do we actually have multiple cores?

Reply:
Yes, I tried and the test passes! 
I had a submit to check if it works.

I did this experiment after checking here: https://github.com/grosser/parallel_tests/wiki/Distributed-Parallel-Tests-and-Travis-Support

It mentions that : 
> Travis recommends using 2 processes for their 1.5 cores per box.

Reply:
Resolving for now.